File: reduce.js
import { TextLineStream } from "jsr:@std/streams/text-line-stream";

async function loadDict(threshold) {
  const dict = [];
  const file = await Deno.open("jieba/extra_dict/dict.txt.big");
  const lineStream = file.readable
    .pipeThrough(new TextDecoderStream())
    .pipeThrough(new TextLineStream());
  for await (const line of lineStream) {
    const arr = line.split(" ");
    const word = arr[0];
    const count = parseInt(arr[1]);
    if (count > threshold) {
      dict.push([word, count]);
    }
  }
  dict.sort((a, b) => {
    if (a[1] < b[1]) return 1;
    if (a[1] > b[1]) return -1;
    return 0;
  });
  return dict;
}

const threshold = 99;
const dict = await loadDict(threshold);
const output = dict.map((x) => x.join(",")).join("\n");
Deno.writeTextFileSync("words.lst", output);

File: reduce.py
words = {}
with open("siminym-zh-repo/all.lst") as f:
    for line in f:
        word = line.split(",", 1)[0]
        words[word] = True

count = 0
with open("cc.zh.300.vec") as f:
    for line in f:
        word = line.split(" ", 1)[0]
        if word in words:
            count += 1

with open("cc.zh.300-small.vec", "w") as fout:
    fout.write(str(count) + " 300\n")
    with open("cc.zh.300.vec") as fin:
        for line in fin:
            word = line.split(" ", 1)[0]
            if word in words:
                fout.write(line)
